samples/src/main/java/com/microsoft/applicationinsights/sample/plugins/GitBuildInfoContextInitializer.java

```diff
@@ -0,0 +1,99 @@
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.sample.plugins;
+
+import java.io.InputStream;
+import java.util.Properties;
+
+import com.microsoft.applicationinsights.extensibility.ContextInitializer;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.telemetry.TelemetryContext;
+
+/**
+ * An initializer to fetch Git data from a properties file
+ * Note: the file with its full package name should appear in the 'ApplicationInsights.xml':
+ *
+ * <ContextInitializers>
+ *   <Add type="com.microsoft.applicationinsights.sample.plugins.GitBuildInfoContextInitializer" />
+ * </ContextInitializers>
+ */
+public class GitBuildInfoContextInitializer implements ContextInitializer {
+    private final static String BUILD_INFO_FILE_NAME = "source-origin.properties";
+
+    final static String GIT_BRANCH_NAME = "git.branch";
+    final static String GIT_COMMIT_NAME = "git.commit";
+    final static String GIT_REPO_NAME = "git.repo";
+    final static String GIT_URL_NAME = "git.url";
+
+    final static String BUILD_DEFAULT_VALUE = "unknown";
+
+    private String gitBranch;
+    private String gitCommit;
+    private String gitRepo;
+    private String gitUrl;
+    private boolean hasBuildData;
+
+    public GitBuildInfoContextInitializer() {
+        try {
+            InputStream inputStream = getClass().getClassLoader().getResourceAsStream(BUILD_INFO_FILE_NAME);
+            if (inputStream == null) {
+                hasBuildData = false;
+                return;
+            }
+
+            Properties buildProperties = new Properties();
+            buildProperties.load(inputStream);
+
+            gitRepo = buildProperties.getProperty(GIT_REPO_NAME, BUILD_DEFAULT_VALUE);
+            gitBranch = buildProperties.getProperty(GIT_BRANCH_NAME, BUILD_DEFAULT_VALUE);
+            gitCommit = buildProperties.getProperty(GIT_COMMIT_NAME, BUILD_DEFAULT_VALUE);
+            if (!gitRepo.equals(BUILD_DEFAULT_VALUE) &&
+                !gitCommit.equals(BUILD_DEFAULT_VALUE)) {
+
+                int index = gitRepo.indexOf(".git");
+                if (index != -1) {
+                    gitUrl = gitRepo.substring(0, index) + '/' + gitCommit;
+                }
+            }
```

Comment:
Shouldn't you have an else part to set `gitRepo` to `UNKNOWN`?
Alternatively (and cleaner) you can initiate `gitRepo` to `UNKNOWN` in which case you can later skip the else part)